### DISCUSSION TITLE
reproducable builds?

### DISCUSSION THREAD

**Discussion by vermaete** [OpenDDS/OpenDDS#4597] [General]
Hi,

Is having the builds of OpenDDS [reproducible](https://reproducible-builds.org/) somewhere planned in one of the future releases?
It could become a desired topic for the audit and quality people.

Somewhere related its that the Yocto build is giving a lot of these [QA](https://docs.yoctoproject.org/singleindex.html#qa-error-and-warning-messages) warnings:
`File <filename> in package <packagename> contains reference to TMPDIR [buildpaths]`
And this because [TAO](https://github.com/DOCGroup/ACE_TAO/blob/5cdada6a08c3dbd5bd496628ab658eb75f4a4d2c/TAO/TAO_IDL/be/be_helper.cpp#L654) is inserting the full path of the IDL file in comment.

Br

**Comment by iguessthislldo** [comment]
I don't think it's been a goal before, but I'm open to it. I'm aware of the concept, but haven't done any research into reproducible builds, so here are my initial thoughts:

- `tao_idl` is also generating a randomized include macro, so that would also have to be controlled if the source code is going to be the same. However does having these in the source code really impact reproducibility? It doesn't change the line numbers, so I wouldn't think these would even change a debug binary. I haven't tested this though, so I could be wrong.

- The only other things I could think of possibly impacting reproducibility would be the `TypeObject`s embedded with type support. The order of items should be stable, but this would have to checked and possibly tested if they're not already.

- How far are reproducible builds supposed to go? Just the core libraries and executables? Tests? Type support libraries generated from user IDL?
